modules/vstudio/vstudio.lua

```diff
@@ -279,8 +282,12 @@
 			arch = iif(isnative, "x86", "Any CPU")
 		end
 
-		if win32 and isnative and arch == "x86" then
-			arch = "Win32"
+		if cfg.system == p.WINDOWS or cfg.system == p.UWP then
```

Comment:
I don't think this really helps fix the problem with the `win32` override. I'd rather see this condition used instead of the `true` that is currently used when calling `archFromConfig` or whichever function has the `true` value. Or, just remove the `win32` argument altogether and maybe replace it with a function that can be overridden as required but the initial impl is just `return cfg.system == p.WINDOWS or cfg.system == p.UWP`.

Ultimately, we still want this module to be compatible with being overridden even if Android and Linux support are brought into it instead of being separate modules.

Reply:
The reason I haven't been able to remove the win32 boolean is that some of the unit tests still need fixing and it's non-trivial for me and out of the scope of this I feel to address it. So I fix the most egregious part (it breaks the other platforms) and I'll create an issue to fix it properly

Reply:
That's fair enough. Happy for this conversation to be resolved when you're ready.